torch/testing/_internal/opinfo/definitions/nested.py

```diff
@@ -205,6 +205,27 @@ def _f(op, sample):
 
 
 # === BEGIN OP-SPECIFIC SAMPLE INPUTS FUNCS ===
+def sample_inputs_clone(op_info, device, dtype, requires_grad, **kwargs):
+    # non-contiguous NJTs
+    for njt in _sample_njts(
+        device=device, dtype=dtype, requires_grad=requires_grad, dims=[2, 3, 4]
+    ):
+        yield SampleInput(njt)
+
+    for memory_format in (torch.contiguous_format, torch.preserve_format):
+        # construct a "non-contiguous with holes" NJT
+        values = torch.randn(
+            10, 5, device=device, dtype=dtype, requires_grad=requires_grad
+        )
+        offsets = torch.tensor([0, 2, 4, 10], device=device, dtype=torch.int64)
+        lengths = torch.tensor([2, 1, 3], device=device, dtype=torch.int64)
+        njt = torch.nested.nested_tensor_from_jagged(
+            values, offsets=offsets, lengths=lengths
+        )
+
+        yield SampleInput(njt, kwargs={"memory_format": memory_format})
```

Comment:
Just to double check, would this test fail even if we didn't properly handle the contiguous_format via unbinding?
Maybe I'm missing something, but it seems like when we call `assertEqualIgnoringNestedInts`, it seems like we would end up unbinding anyway.

Reply:
yep, I verified that the test fails before my `contiguous_format` handling logic. The reason why is that before this PR, the lengths metadata wouldn't be propagated and so unbinding would just give you the wrong results (not taking into account lengths).

Reply:
I see, thanks